x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/parser/EscapedFunctionsTests.java

```diff
@@ -62,6 +62,40 @@ private Literal timestampLiteral(String date) {
         return (Literal) exp;
     }
 
+    private String buildDate() {
+        StringBuilder sb = new StringBuilder();
+        int length = randomIntBetween(4, 9);
+
+        if (randomBoolean()) {
+            sb.append('-');
+        } else {
+            if (length > 4) {
+                sb.append('-');
+            }
+        }
+
+        for (int i = 1; i <= length; i++) {
+            sb.append(i);
+        }
+        sb.append("-05-10");
+        return sb.toString();
+    }
+
+    private String buildSecsAndFractional() {
+        boolean hasSecs = randomBoolean();
+        String secs = "";
+        if (hasSecs) {
+            secs = ":55";
+        }
+
+        String fractionalSecs = "";
+        if (hasSecs) {
+            fractionalSecs = randomFrom("", ".1", ".12", ".123", ".1234", ".12345", ".123456",
+                    ".1234567", ".12345678", ".123456789");
+        }
+        return secs + fractionalSecs;
```

Comment:
I think this whole code could be simplified (and still readable and not cluttered) as:
```
        if (randomBoolean()) {
           return ":55" + randomFrom("", ".1", ".12", ".123", ".1234", ".12345", ".123456",
                    ".1234567", ".12345678", ".123456789");
        }
        return "";
```